### ISSUE TITLE
Get buffer from mapnik.Image

### ISSUE THREAD

**Issue by StevenLooman** [mapnik/pymapnik2#6]
`mapnik.Image` has a `save()` method which can be used to save an image to a given filename, for example, the following will save the image to a file called `tmp.png` in the `png256` format:

```
im = mapnik.Image(255,267)
im.background = mapnik.Color('rgba(1,2,3,.5)')
im.save('tmp.png', 'png256')
```

When one wants to use the data from the image (variable `im` in the upper example), one has to open the just-saved file and read the data.

Is it possible to save the image in a given format to an in-memory buffer? I noticed that  the `mapnik.Image` wrapper does have the `tostring()` method, but this returns a string encoded in some sort of a hex format. I'm sure this can be decoded, but it would be easier to directory save the image to a buffer.

I'm unfamiliar with the mapnik internals, but if wanted, I can investigate and probably provide a patch for pymapnik2 to support this.

**Comment by StevenLooman** [comment]
I guess the function `save_to_string` at https://github.com/mapnik/mapnik/blob/master/src/image_util.cpp#L96 should be used?

**Comment by StevenLooman** [comment]
Easily fixed, tested with Python 2.7:

```
from StringIO import StringIO
string = image.tostring('png256')
buffer = StringIO(string)
```

No need for any changes to pymapnik2.
